examtool/examtool/api/utils.py

```diff
@@ -28,3 +24,39 @@ def dict_to_list(d):
 
 def list_to_dict(l):
     return {i: x for i, x in enumerate(l)}
+
+
+class IDFactory:
+    def __init__(self, id_start="", id_end="", length=32, allow_random_ids=True):
+        self.length = length
+        self.id_start = id_start
+        self.id_end = id_end
+        self.allow_random_ids = allow_random_ids
+        self.current_ids = set()
+
+    def rand_id(self, length=None):
```

Comment:
Keep this as a separate function - it doesn't need to be a method. Then you don't have to change where it's used elsewhere. Just add the optional length argument.

Reply:
Okay I moved it back out.